src/main/java/org/folio/rest/helper/TransactionsHelper.java

```diff
@@ -67,4 +74,36 @@ private Transaction modifyTransaction(Transaction transaction, AwaitingPayment a
     return transaction;
   }
 
+  public CompletableFuture<Transaction> createEncumbrance(Transaction encumbrance) {
+    return checkEncumbranceRestrictions(encumbrance)
+      .thenCompose(v -> createTransaction(encumbrance));
+  }
+
+  public CompletableFuture<Void> checkEncumbranceRestrictions(Transaction encumbrance) {
+    String budgetQuery = "fundId==" + encumbrance.getFromFundId() + " AND fiscalYearId==" + encumbrance.getFiscalYearId();
+    return new BudgetsHelper(httpClient, okapiHeaders, ctx, lang).getSingleBudgetByQuery(budgetQuery)
+      .thenCompose(budget -> {
+        if (budget.getBudgetStatus() == Budget.BudgetStatus.ACTIVE) {
+          String ledgerQuery = "fund.id==" + encumbrance.getFromFundId() + " AND fiscalYear.id==" + encumbrance.getFiscalYearId();
```

Comment:
Ledger not depends on fiscalYear, so you only need a fund.id to find the related ledger.